client/my-sites/marketplace/components/reviews-list/index.tsx

```diff
@@ -104,21 +129,71 @@ export const MarketplaceReviewsList = ( props: MarketplaceReviewsQueryProps ) =>
 								{ moment( review.date ).format( 'll' ) }
 							</div>
 						</div>
-
-						<div
-							// sanitized with sanitizeSectionContent
-							// eslint-disable-next-line react/no-danger
-							dangerouslySetInnerHTML={ {
-								__html: sanitizeSectionContent( review.content.rendered ),
-							} }
-							className="marketplace-reviews-list__content"
-						></div>
+						{ isEditing && review.author === currentUserId ? (
+							<>
+								<div className="marketplace-reviews-list__review-rating">
+									<h2>{ translate( 'Let us know how your experience has changed' ) }</h2>
+									<ReviewsRatingsStars
+										size="medium-large"
+										rating={ editorRating }
+										onSelectRating={ setEditorRating }
+									/>
+								</div>
+								<TextareaControl
+									rows={ 4 }
+									cols={ 40 }
+									name="content"
+									value={ editorContent }
+									onChange={ setEditorContent }
+								/>
+							</>
+						) : (
+							<div
+								// sanitized with sanitizeSectionContent
+								// eslint-disable-next-line react/no-danger
+								dangerouslySetInnerHTML={ {
+									__html: sanitizeSectionContent( review.content.rendered ),
+								} }
+								className="marketplace-reviews-list__content"
+							></div>
+						) }
 						<div className="marketplace-reviews-list__review-actions">
-							{ review.author === currentUserId && (
+							{ isEditing && review.author === currentUserId && (
+								<>
+									<CheckboxControl
+										className="marketplace-reviews-list__checkbox"
+										label={ translate( 'Notify me when my review is approved and published.' ) }
```

Comment:
Let's hide this behind a feature flag so that we can ship the editing functionality right away.